qiskit/quantum_info/states/densitymatrix.py

```diff
@@ -585,6 +585,24 @@ def _evolve_instruction(self, obj, qargs=None):
         vec._append_instruction(obj, qargs=qargs)
         return vec
 
+    def to_statevector(self):
+        """Return a statevector from a pure density matrix.
+
+        Returns:
+            Statevector: The pure density matrix's corresponding statevector.
+                Corresponds to the eigenvector of the only non-zero eigenvalue.
+
+        Raises:
+            QiskitError: if the state is not pure.
+        """
+        if not np.isclose(self.purity(), 1):
+            raise QiskitError(
+                    "Cannot obtain statevector from non-pure density matrix.")
+
+        eigvals, eigvecs = np.linalg.eig(self.data)
+        psi = eigvecs[:, np.argmax(eigvals)]  # eigenvector are in columns.
+        return Statevector(psi)
```

Comment:
there should only be one  eigenvalue  and  you should check if it is one.  I thought there was a command to get the eigen decomposition (well i  used   it in qsphere) which would be better to use  and only accept if it has  one eigenvector and  equal  to one